woocommerce/google-listings-and-ads
# Add `eslint-plugin-import` to help catch JS import errors

### Changes proposed in this Pull Request:

This PR serves as a workaround to address my comment in https://github.com/woocommerce/google-listings-and-ads/pull/899#issuecomment-885033942. It adds [`eslint-plugin-import`](https://www.npmjs.com/package/eslint-plugin-import) and uses the recommended config to help us catch import errors in JavaScript side. 

Note that the file `.eslintrc` has also been renamed to `.eslintrc.js`. This is consistent with Gutenberg and WC Admin.

The approach used in this PR is also used in WC Admin, where they have manual install `eslint-plugin-import` package. We are using `plugin:import/recommended` config, whereas they are using manual individual rules configuration. See links below:

https://github.com/woocommerce/woocommerce-admin/blob/0f6eafcfa3dfbc4c7807210ef1056440d896654d/package.json#L213

https://github.com/woocommerce/woocommerce-admin/blob/0f6eafcfa3dfbc4c7807210ef1056440d896654d/.eslintrc.js

### Screenshots:

Before this PR, if you import something that doesn't exist, there is no error and warning, and might lead to runtime error in production:

![image](https://user-images.githubusercontent.com/417342/126833657-74085a3c-9a5c-491d-9ebf-9d52cfa48541.png)

With this PR, there will be eslint error:

![image](https://user-images.githubusercontent.com/417342/126833777-81365dd1-695d-467e-856c-d311d43e545a.png)

### Detailed test instructions:

1. Checkout this PR. 
2. Open a JS file and import something that does not exist, e.g. `import { Button } from '@wordpress/element';`
3. There should be an ESLint error on the wrong import.

### Changelog entry

> Fix: add eslint-plugin-import to help catch JavaScript import errors.
